client/src/Components/Results.js

```diff
@@ -126,44 +130,179 @@ export default function Results(props) {
 		setAttemptNumber(attemptNumber - 1);
 	};
 
+	const getScore = (student) => {
+		let studentScores = 0;
+		if (quizQuestions.length > 0) {
+			quizQuestions.map((question) => {
+				studentScores = studentScores + scoreCounter(student, question._id);
+			});
+			return studentScores;
+		}
+	};
+
+	//calculat the scores
+	const scoreCounter = (student, questionId) => {
+		const allAttempts = quizSelectedResults.filter(
+			(quizResult) =>
+				quizResult.studentName === student
+        && quizResult.question_id === questionId
+		);
+		if (allAttempts.length <= 0) {
+			return 0;
+		}
+
+		allAttempts.length > attemptCounter
+			? setAttemptCounter(allAttempts.length)
+			: null;
+		let timestamp = getLastAttempt(student);
+		if (!timestamp) {
+			return null;
```

Comment:
Is it a problem to return `null` here instead of 0?

Reply:
good feedback
It should return 0